core/src/main/scala/akka/persistence/cassandra/query/EventsByTagStage.scala

```diff
@@ -577,18 +581,31 @@ import com.datastax.driver.core.utils.UUIDs
               m.copy(queryPrevious = false, bucket = m.bucket.previous(1)))))
             lookForMissing()
           } else {
-            log.debug(
-              "[{}] [{}]: Still looking for missing. {}. Waiting for next poll.",
-              stageUuid,
-              session.tag,
-              stageState)
-            updateStageState(_.copy(missingLookup = stageState.missingLookup.map(m => {
-              val newBucket = TimeBucket(m.maxOffset, bucketSize)
-              m.copy(bucket = newBucket, queryPrevious = !newBucket.within(m.previousOffset))
-            })))
-            // a current query doesn't have a poll schedule one
-            if (!isLiveQuery())
-              scheduleOnce(QueryPoll, settings.refreshInterval)
+            val timeLeft = missing.deadline.timeLeft
+            if (timeLeft <= Duration.Zero) {
+              abortMissingSearch(missing)
```

Comment:
^ part one of the fix

Reply:
It depends what kind of missing search is happening.

If it is a new persistence id then it is set to the usually small `new-persistence-id-scan-timeout` from `handleFirstTimePersistenceId`

If it is an actual gap, e.g. we see 1-4, then say 6 then it is set to the, much larger as we will fail the stream, `gap-timeout` set in `handleExistingPersistenceId`

Reply:
Ok, but abortMissingSearch will fail the stream also for the first case (handleFirstTimePersistenceId), or did I misread that?

Reply:
`abortMissingSearch` only fails the stream if `failIfNotFound` is set to true, which is only set in the second case.

I'll re-name that field in `LookingForMissing` to `gapDetected` to make it clearer (I hope).